server/portal_server.js

```diff
@@ -63,18 +74,19 @@ app.get('/assets/download/:id', (req, res) => {
           'Content-Type': 'application/octet-stream',
           'Content-Length': contentLength
         });
-        let file_stream = storage.get(id);
-
-        file_stream.on(notLocalHost ? 'finish' : 'close', () => {
-          storage.forceDelete(id).then(err => {
-            if (!err) {
-              console.log('Deleted.');
-            }
-          });
-        });
+      });
```

Comment:
the stream pipe needs to happen after the writeHead otherwise the pipe can cause the default headers to get sent and cause writeHead to throw an error.